fastly/object_store.go

```diff
@@ -0,0 +1,214 @@
+package fastly
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// https://developer.fastly.com/reference/api/object-store/
+
+// TODO(dgryski): Should the store id be `StoreID` or `ID` consistently?
+
+type ObjectStore struct {
+	Name      string     `mapstructure:"name"`
+	ID        string     `mapstructure:"id"`
+	CreatedAt *time.Time `mapstructure:"created_at"`
+	UpdatedAt *time.Time `mapstructure:"updated_at"`
+}
+
+type CreateObjectStoreInput struct {
+	Name string `json:"name"`
+}
+
+func (c *Client) CreateObjectStore(i *CreateObjectStoreInput) (*ObjectStore, error) {
+	if i.Name == "" {
+		return nil, ErrMissingName
+	}
+
+	const path = "/resources/stores/object"
+	resp, err := c.PostJSON(path, i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var store *ObjectStore
+	if err := decodeBodyMap(resp.Body, &store); err != nil {
+		return nil, err
+	}
+	return store, nil
+}
+
+type ListObjectStoresResponse struct {
+	Data []ObjectStore
+	Meta map[string]string
+}
+
+func (c *Client) ListObjectStores() (*ListObjectStoresResponse, error) {
+	const path = "/resources/stores/object"
+	// TODO(dgryski): How do I handle pagination?
```

Comment:
I'd say look to https://github.com/fastly/go-fastly/blob/0092ed3705eca1fe2bbcca43a863c9d49471e626/fastly/dictionary_item.go#L72 for an example implementation.

Original PR: https://github.com/fastly/go-fastly/pull/321

Reply:
iirc I had suggested maybe refactoring the paginator once we got to go 1.18 because having to recreate the paginator logic across types could be tedious, but A.) go-fastly won't support 1.18 anytime soon, and B.) even if it did I can't recall if it's possible to abstract with generics.